__tests__/events.ts

```diff
@@ -0,0 +1,264 @@
+          username: testUserInfo.username,
+        },
+        start: testEventInfo.start.toISOString(),
+        end: testEventInfo.end.toISOString(),
+        venue: testEventInfo.venue,
+        image_url: "",
+        description: testEventInfo.description,
+      },
+    });
+
+    await deleteEvents();
+    await deleteUsers();
+  });
+});
+
+describe("event deletion", () => {
+  test("should be able to delete an event", async () => {
+    // Create user in the database
+    const user: User = await createUser(testUserInfo);
+    const testServer = await createTestServerWithUserLoggedIn(user);
+    // Create a test client connected to the test server
+    const client = createTestClient(testServer);
+    // Create a test event
+    const event: Event = await createEvent(testEventInfo);
+
+    const mutation = gql`
+      mutation($id: ID!) {
+        deleteEvent(id: $id) {
+          id
+        }
+      }
+    `;
+    const response: GraphQLResponse = await client.mutate({
+      mutation,
+      variables: { id: event.id },
+    });
+    expect(response.data).toEqual({
+      deleteEvent: {
+        id: event.id,
+      },
+    });
+
+    await deleteEvents();
+    await deleteUsers();
+  });
+});
+
+describe("event subscriber addition", () => {
+  test("should be able to add current user to event subscribers list", async () => {
+    // Create user in the database
+    const testUser: User = await createUser(testUserInfo);
+    const testServer = await createTestServerWithUserLoggedIn(testUser);
+    // Create a test client connected to the test server
+    const client = createTestClient(testServer);
+    // Create a test event
+    const { id: event_id }: Event = await createEvent(testEventInfo);
+
+    const mutation = gql`
+      mutation($id: ID!) {
+        addEventSubscriber(id: $id) {
+          subscribers {
+            id
+          }
+        }
+      }
+    `;
+
+    const {
+      data: { addEventSubscriber },
+    }: GraphQLResponse = await client.mutate({
+      mutation,
+      variables: { id: event_id },
+    });
+    const { subscribers } = addEventSubscriber;
+    expect(subscribers).toEqual(expect.arrayContaining([{ id: testUser.id }]));
+
+    await deleteEvents();
+    await deleteUsers();
+  });
+});
+
+describe("event subscriber removal", () => {
+  test("should be able to remove current user from event subscribers list", async () => {
+    // Create user in the database
+    const testUser: User = await createUser(testUserInfo);
+    const testServer = await createTestServerWithUserLoggedIn(testUser);
+    // Create a test client connected to the test server
+    const client = createTestClient(testServer);
+    // Create a test event
+    const { id: event_id }: Event = await createEvent(testEventInfo);
+
+    const addSubscriberMutation = gql`
+      mutation($id: ID!) {
+        addEventSubscriber(id: $id) {
+          subscribers {
+            id
+          }
+        }
+      }
+    `;
```

Comment:
instead of using a `graphql mutation` to add a subscriber, use `prisma` to `updateEvent` with a subscriber list first, then test the `graphql delete mutation`

the key point is to test each mutation independently from one another

Reply:
ohh rightt, let me change thatt